iguagile/client.go

```diff
@@ -78,3 +82,81 @@ func (c *Client) Send(message []byte) {
 func (c *Client) Close() error {
 	return c.conn.Close()
 }
+
+// ClientManager manages clients.
+type ClientManager struct {
+	clients map[int]*Client
+	*sync.Mutex
+}
+
+// NewClientManager is ClientManager constructed.
+func NewClientManager() *ClientManager {
+	return &ClientManager{
+		clients: make(map[int]*Client),
+		Mutex:   &sync.Mutex{},
+	}
+}
+
+// Get the client.
+func (m *ClientManager) Get(clientID int) (*Client, error) {
+	m.Lock()
+	client, ok := m.clients[clientID]
+	m.Unlock()
+	if !ok {
+		return nil, fmt.Errorf("client not exists %v", clientID)
+	}
+
+	return client, nil
+}
+
+// Add the client.
+func (m *ClientManager) Add(client *Client) error {
+	m.Lock()
+	defer m.Unlock()
+
+	if _, ok := m.clients[client.GetID()]; ok {
+		return fmt.Errorf("client already exists %v", client.GetID())
+	}
+
+	m.clients[client.GetID()] = client
+	return nil
+}
+
+// Remove the client.
+func (m *ClientManager) Remove(clientID int) {
+	m.Lock()
+	defer m.Unlock()
+
+	if _, ok := m.clients[clientID]; !ok {
+		return
+	}
+
+	delete(m.clients, clientID)
+}
+
+// Exist checks the client exists.
+func (m *ClientManager) Exist(clientID int) bool {
+	m.Lock()
+	_, ok := m.clients[clientID]
+	m.Unlock()
+	return ok
+}
+
+// GetClientsMap returns all clients.
+func (m *ClientManager) GetClientsMap() map[int]*Client {
```

Comment:
```suggestion
func (m *ClientManager) GetAllClients() map[int]*Client {
```
ですかね。メソッドに型情報があるのは微妙なので